CharlieBackend.Api/Controllers/AccountsController.cs

```diff
@@ -104,7 +104,7 @@ public async Task<ActionResult> SignIn([FromBody]AuthenticationDto authenticatio
                     return BadRequest();
                 }
 
-                entityId = foundMentor.Id;
+                entityId = foundMentor.Data.Id;
             }
 
             if (foundAccount.Role == UserRole.Secretary)
```

Comment:
also please make same check here, although SecretaryService.GetSecretaryByAccountIdAsync assumes that it's always a secretary record for account in secretary role.